playground/backend/internal/preparers/python_preparers.go

```diff
@@ -112,3 +133,76 @@ func writeToFile(to *os.File, str string) error {
 	}
 	return nil
 }
+
+// saveGraph adds code to pipeline to save the pipeline's graph to the file GraphFileName
+func saveGraph(from *os.File, to *os.File) error {
+	newLine := false
+	reg := regexp.MustCompile(findPipelinePattern)
+	scanner := bufio.NewScanner(from)
+	pipelineName := ""
+	spaces := ""
+	err := errors.New("")
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if pipelineName == "" {
+			// Try to find where the beam pipeline declaration is located
+			err = writeLineToFile(findPipelineLine)(newLine, to, &line, &spaces, &pipelineName, &reg)
+		} else if reg != nil {
+			// Try to find where beam pipeline definition is finished and add code to store the graph
+			reg = regexp.MustCompile(fmt.Sprintf(indentationPattern, spaces))
+			err = writeLineToFile(addCodeForGraph)(newLine, to, &line, &spaces, &pipelineName, &reg)
+		} else {
+			err = writeLineToFile(func(line, spaces, pipelineName *string, reg **regexp.Regexp) {
+				// No need to find or change anything, just write current line to file
+			})(newLine, to, &line, &spaces, &pipelineName, nil)
+		}
+		if err != nil {
+			logger.Errorf("Preparation: Error during write \"%s\" to tmp file, err: %s\n", line, err.Error())
+			return err
+		}
+		newLine = true
+	}
+	return scanner.Err()
```

Comment:
This "if" combination really confuses me. Could we simplify it? For example, make writeLineToFile simpler to follow the Single Responsibilities principle? 
It also will be great to have documentation for complex functions.

Reply:
Done